### PR TITLE
SW-5122 Create header for participant home page

### PR DESCRIPTION
**Pull request by nickgraz**
- Add current module and current participant project to `ParticipantProvider`

### CODE REVIEW

**Comment by nickgraz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2515?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2515** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2515?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2514** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2514?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2513** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2513?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by karthikbtf — APPROVED** [review]
Nice

File: src/providers/Participant/ParticipantProvider.tsx [hunk 1]
```diff
@@ -17,32 +21,65 @@ const ParticipantProvider = ({ children }: Props) => {
...
+
+  const participant = useAppSelector(selectParticipant(currentParticipantProject?.participantId || -1));
+  const currentModule = useAppSelector(selectModule(participant?.currentModuleId || -1));
   const projects = useAppSelector(selectProjects);
 
-  // State for provider data
-  const [participantProjects, setParticipantProjects] = useState<Project[]>([]);
+  const _setCurrentParticipantProject = useCallback(
+    (projectId: string | number) => {
+      setCurrentParticipantProject(participantProjects.find((project) => project.id === Number(projectId)));
```
[the remark this answers is not in the document]
Comment (nickgraz) [reply src/providers/Participant/ParticipantProvider.tsx:33]: `useState` bakes in `| undefined` to the supplied type. We can add it but it is redundant

**Comment by nickgraz** [comment]
### Merge activity

* **Apr 16, 11:35 AM EDT**: @nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2515).

**Pull request merged**